tasks/abs.rb

```diff
@@ -120,13 +126,14 @@ def tear_down(node_name, inventory_location)
 
 params = JSON.parse(STDIN.read)
 platform = params['platform']
+platforms = params['platforms']
 action = params['action']
 node_name = params['node_name']
 inventory_location = sanitise_inventory_location(params['inventory'])
 vars = params['vars']
 raise 'specify a node_name when tearing down' if action == 'tear_down' && node_name.nil?
-raise 'specify a platform when provisioning' if action == 'provision' && platform.nil?
-unless node_name.nil? ^ platform.nil?
+raise 'specify a platform when provisioning' if action == 'provision' && platforms.nil?
+unless node_name.nil? ^ platforms.nil?
```

Comment:
with the above change this is now the correct error message and validation logic